examples/run/cuda/seeding_example_cuda.cpp

```diff
@@ -6,79 +6,165 @@
     vecmem::cuda::device_memory_resource device_mr;
     traccc::memory_resource mr{device_mr, &cuda_host_mr};
 
-    // Read the surface transforms
-    traccc::geometry surface_transforms;
-    if (i_cfg.run_detray_geometry == false) {
-        surface_transforms = traccc::io::read_geometry(i_cfg.detector_file);
-    } else if (i_cfg.run_detray_geometry == true) {
-        // Declare detector type
-        using detector_t = detray::detector<detray::toy_metadata<>>;
-
-        // Read the detector
-        detray::io::detector_reader_config reader_cfg{};
-        reader_cfg.add_file(traccc::io::data_directory() + i_cfg.detector_file)
-            .add_file(traccc::io::data_directory() + i_cfg.material_file);
+    // Performance writer
+    traccc::seeding_performance_writer sd_performance_writer(
+        traccc::seeding_performance_writer::config{});
+    traccc::finding_performance_writer find_performance_writer(
+        traccc::finding_performance_writer::config{});
+    traccc::fitting_performance_writer fit_performance_writer(
+        traccc::fitting_performance_writer::config{});
 
-        const auto [det, names] =
-            detray::io::read_detector<detector_t>(host_mr, reader_cfg);
+    traccc::nseed_performance_writer nsd_performance_writer(
+        "nseed_performance_",
+        std::make_unique<traccc::simple_charged_eta_pt_cut>(
+            2.7f, 1.f * traccc::unit<traccc::scalar>::GeV),
+        std::make_unique<traccc::stepped_percentage>(0.6f));
 
-        surface_transforms = traccc::io::alt_read_geometry(det);
+    if (common_opts.check_performance) {
+        nsd_performance_writer.initialize();
     }
 
     // Output stats
     uint64_t n_modules = 0;
     uint64_t n_spacepoints = 0;
     uint64_t n_seeds = 0;
     uint64_t n_seeds_cuda = 0;
+    uint64_t n_found_tracks = 0;
+    uint64_t n_found_tracks_cuda = 0;
+    uint64_t n_fitted_tracks = 0;
+    uint64_t n_fitted_tracks_cuda = 0;
+
+    /*****************************
+     * Build a geometry
+     *****************************/
 
-    // Configs
+    // B field value and its type
+    // @TODO: Set B field as argument
+    const traccc::vector3 B{0, 0, 2 * detray::unit<traccc::scalar>::T};
+
+    host_detector_type host_det{mng_mr};
+
+    // Read the surface transforms
+    traccc::geometry surface_transforms;
+
+    if (common_opts.run_detray_geometry == false) {
+        surface_transforms =
+            traccc::io::read_geometry(common_opts.detector_file);
+    } else if (common_opts.run_detray_geometry == true) {
+
+        // Read the detector
+        detray::io::detector_reader_config reader_cfg{};
+        reader_cfg
+            .add_file(traccc::io::data_directory() + common_opts.detector_file)
+            .add_file(traccc::io::data_directory() + common_opts.material_file)
+            .bfield_vec(B[0], B[1], B[2]);
+
+        auto [det, names] =
+            detray::io::read_detector<host_detector_type>(host_mr, reader_cfg);
```

Comment:
Btw, should it not be:
```suggestion
            detray::io::read_detector<host_detector_type>(mng_mr, reader_cfg);
```

Reply:
Maybe it was right. what should be called with `mng_mr` is `host_det`